sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -10,12 +10,22 @@
 namespace Azure
 {
     /// <summary>
-    /// Options which can be used to control the behavior of a request sent by a client.
+    /// Options that can be used to control the behavior of a request sent by a client.
     /// </summary>
     public class RequestContext
     {
+        private HttpMessageClassifier[]? _classifiers;
+        private bool _frozen;
+
         internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
 
+        internal List<(int Status, bool IsError)>? StatusCodes { get; private set; }
+
+        internal HttpMessageClassifier[]? MessageClassifiers
```

Comment:
It was because you can't pass a property as a ref parameter to Array.Resize().  But if we make the StatusCodes List an array as well, we'll need backing fields for both of them, so at least have consistency.